### ISSUE TITLE
Errors in pimatic 0.8.99

### ISSUE THREAD

**Issue by sweebee** [sirchia/pimatic-rflink#4]
I'm getting a lot of errors since the last update of pimatic:

```
22:47:43.268 [pimatic] info: Loading plugin: "pimatic-rflink" (0.4.0)
22:47:45.665 [pimatic] error: Invalid config of pimatic-rflink: Missing property "reconnectInterval"
error [pimatic]: Invalid config of device "woonkamer-scenes": Missing property "required" in /protocols/0, Missing property "required" in /protocols/1
22:43:18error [pimatic]: Invalid config of device "keuken-tafel-switch": Missing property "required" in /protocols/0
22:43:18error [pimatic]: Invalid config of device "woonkamer": Missing property "required" in /protocols/0, Missing property "required" in /protocols/1
22:43:17error [pimatic]: Invalid config of device "keuken": Missing property "required" in /protocols/0
22:43:17error [pimatic]: Invalid config of device "woonkamer-plafond": Missing property "options" in /protocols/0, Property "id" is not a valid property in /protocols/0, Property "switch" is not a valid property in /protocols/0
22:43:17error [pimatic]: Invalid config of device "buiten-water": Missing property "required" in /protocols/0, Missing property "required" in /protocols/1
22:43:17error [pimatic]: Invalid config of device "overloop-lamp": Missing property "required" in /protocols/0
22:43:17error [pimatic]: Invalid config of device "meterkast-lamp": Missing property "required" in /protocols/0
22:43:17error [pimatic]: Invalid config of device "bijkeuken-lamp": Missing property "required" in /protocols/0
22:43:17error [pimatic]: Invalid config of device "deurbel": Missing property "required" in /protocols/0
22:43:17error [pimatic]: Invalid config of device "bijkeuken-pir": Missing property "required" in /protocols/0
22:43:17error [pimatic]: Invalid config of device "hal-lamp": Missing property "required" in /protocols/0
22:42:55error [pimatic]: Invalid config of pimatic-rflink: Missing property "reconnectInterval"
```

**Comment by sweebee** [comment]
Still got an error with a dimmer, can't find it:

```
23:05:11.480 [pimatic] Invalid config of device "woonkamer-plafond": Missing property "options" in /protocols/0, Property "id" is not a valid property in /protocols/0, Property "switch" is not a valid property in /protocols/0
```

**Comment by sweebee** [comment]
Fixed the dimmer also. Atleast no errors anymore.

**Comment by sirchia** [comment]
It seems that the Pimatic devs made a backwards incompatible change to the way configuration schema's are validated, more specificaly: how required attributes are declared. The change doesn't seem too daunting though, so will see if I can fix it after work.

**Comment by sweebee** [comment]
This should work?

https://github.com/pimatic/pimatic/blob/master/device-config-schema.coffee#L56

**Comment by sirchia** [comment]
Should be fixed in 0.4.1.
